File: src/lira/linguistics/role/prompt_tokenizer.py
"""Top-level entry point: wraps a UserPrompt and drives GraphProcessor
to build its full Subject tree."""

from .graph_processor import GraphProcessor
from ..data.units import LinguisticUnitKind, Subject, UserPrompt


class PromptTokenizer:
    def __init__(self, graph_processor: GraphProcessor):
        self.graph_processor = graph_processor

    def tokenize_prompt(self, prompt: UserPrompt) -> Subject:
        prompt.system_property = self.graph_processor.create_property_wrapper(
            prompt, LinguisticUnitKind.UserPrompt, 0, "PromptTokenizer_Gateway"
        )
        return self.graph_processor.process_subject(prompt.text, 0)

File: prototype/src/lira/linguistics/role/prompt_tokenizer.ts
import { LinguisticUnitKind } from "../data/linguistic_unit_kind";
import type { Document } from "../data/document";
import type { UserPrompt } from "../ui/user_prompt";
import type { GraphProcessor } from "./graph_processor";

/** Top-level entry point: wraps a UserPrompt and drives GraphProcessor
 * to build its full Document tree.
 *
 * Ported from linguistics/role/prompt_tokenizer.py. */
export class PromptTokenizer {
  constructor(private readonly graphProcessor: GraphProcessor) {}

  tokenizePrompt(prompt: UserPrompt): Document {
    prompt.systemProperty = this.graphProcessor.createPropertyWrapper(
      prompt, LinguisticUnitKind.UserPrompt, 0, "PromptTokenizer_Gateway",
    );
    return this.graphProcessor.processDocument(prompt.text, 0);
  }
}
